src/uct/tcp/tcp_ep.c

```diff
@@ -135,7 +135,7 @@ static void uct_tcp_ep_cleanup(uct_tcp_ep_t *ep)
         uct_tcp_ep_ctx_reset(&ep->rx);
     }
 
-    if (ep->events && (ep->fd != -1)) {
+    if ((ep->events != 0) && !(ep->flags & UCT_TCP_EP_FLAG_FAILED)) {
```

Comment:
aren't we setting events to 0 when failed?

Reply:
better to have an assert for it